Lesson4/src/Homework4/Services/MilitaryPart.java

```diff
@@ -0,0 +1,106 @@
+package Homework4.Services;
+
+import Homework4.Model.Person;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class MilitaryPart {
+
+
+    private int size;
+    private Person[] sizeMilitary;
+
+
+    public MilitaryPart(List<Person> fitPeople) {
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Can you enter size Military part, please\t ");
+        size = scanner.nextInt();
+        sizeMilitary = new Person[size];
+        for (int i = 0; i < size; i++) {
+            sizeMilitary[i] = new Person(0, "UNNAMED", 0, "men");
+        }
+        int longList = 0;
+        for (Person i : fitPeople) {
+            longList++;
+        }
+        for (int i = 0; i < size; i++) {
+            if (i > longList - 1) {
+                break;
+            } else if (sizeMilitary[i].getName() == "UNNAMED") {
+                this.sizeMilitary[i] = fitPeople.get(i);
+            }
+
+        }
+
+        if (size < longList) {
+            for (int i = longList; i > size; i--) {
+                System.out.println(fitPeople.get(i - 1).getName() + "\tPerson doesn't enter to MilitaryPart");
+            }
+        }
+
+    }
+
+    public void newSoldier(Person person) {
+        int freeSpace = sizeMilitary.length;
+        for (int i = 0; i < sizeMilitary.length; i++) {
+            if (sizeMilitary[i].getName() == person.getName() && sizeMilitary[i].getAddress() == person.getAddress() && sizeMilitary[i].getAge()
+                    == person.getAge()) {
+                System.out.println("We have already taken\t" + person.getName() + "\taway");
+                break;
+
+            }
+            if (sizeMilitary[i].getName() == "UNNAMED") {
+                sizeMilitary[i] = person;
+                break;
```

Comment:
so when you found a place for a soldier, you take it, and then you call break. Because of this freeSpace-- won't be called. So I think you should move line 57 to line 47 (first line in the loop). 

Alternatively, you can make freeSpace a global variable, by default it will be equal to sizeMilitary.length, and then, every time you will add a person to the MilitaryPart - you will decrease it